### DISCUSSION TITLE
M2 Pro Macbook Pro on HDR-enabled external screen - HDR videos are very dim (1.3.1 nightly)

### DISCUSSION THREAD

**Discussion by Tristan42424** [iina/iina#4469] [Q&A]
M2 Pro Macbook Pro on HDR-enabled external screen - HDR videos are very dim, almost completely black except for highlights. 

Engaging Mission Control with touchpad resolves it, but as soon as I let go of the touchpad the issue returns.

Issue remains in full screen and windowed. Sometimes it flashes with the correct brightness but immediately returns to being dim.
![IMG_1771](https://github.com/iina/iina/assets/48087844/1e789504-736b-413c-8b0a-4b34aa3aeaef)
![IMG_1770](https://github.com/iina/iina/assets/48087844/01e9695c-bd85-4ae3-b00b-f7ee2b6ee2d0)

**Comment by low-batt** [comment]
Thanks for reporting. This is issue #4312. The nightly build contains HDR related changes. We are hearing mixed reports on how it behaves on Macs with the M2 chip. Some are reporting HDR is working with the nightly build. Others are still experiencing these problems. Best to continue discussions in issue #4312.

**Reply by zeronullempty** [reply]
Hopefully I'll be able to ditch Infuse player for HDR content on my M1 Pro Macbook Pro once this new version comes out.

**Comment by joonhyuk** [comment]
Same issue here. M2 pro / thunderbolt3 dock / dp1.2 connection
When I play the video with mpv cli(brew installed), HDR plays beautifully.

**Reply by low-batt** [reply]
> When I play the video with mpv cli(brew installed), HDR plays beautifully.

But does it play it with EDR activated?

Download the YouTube video [The World in HDR in 4K (ULTRA HD)](https://www.youtube.com/watch?v=tO01J-M3g0U). Then play it on your MacBook Pro's internal XDR screen with mpv using these options:
```text
mpv --autofit=99% --cocoa-cb-output-csp=bt.2100-pq --target-prim=bt.2020 --target-trc=pq --vo=libmpv ~/Movies/The\ World\ in\ HDR\ in\ 4K\ \(ULTRA\ HD\)\ \[tO01J-M3g0U\].mp4
```

Then repeat playing it exactly the same way, but using the external screen.
